x-pack/plugins/security_solution/server/fleet_integration/notify_protection_feature_usage.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { NewPackagePolicy } from '../../../fleet/common';
+import { PolicyConfig, ProtectionModes } from '../../common/endpoint/types';
+import { EndpointMetadataService } from '../endpoint/services/metadata';
+import { FeatureUsageService } from '../endpoint/services/feature_usage/service';
+
+const OS_KEYS = ['windows', 'mac', 'linux'] as const;
+const PROTECTION_KEYS = ['memory_protection', 'behavior_protection'] as const;
+
+function isNewlyEnabled(current: ProtectionModes, next: ProtectionModes) {
+  if (current === 'off' && (next === 'prevent' || next === 'detect')) {
+    return true;
+  }
+
+  return false;
+}
+
+function notifyProtection(type: string, featureUsageService: FeatureUsageService) {
+  switch (type) {
+    case 'ransomware':
+      featureUsageService.notifyUsage('RANSOMWARE_PROTECTION');
+      return;
+    case 'memory_protection':
+      featureUsageService.notifyUsage('MEMORY_THREAT_PROTECTION');
+      return;
+    case 'behavior_protection':
+      featureUsageService.notifyUsage('BEHAVIOR_PROTECTION');
+  }
+}
+
+export async function notifyProtectionFeatureUsage(
+  newPackagePolicy: NewPackagePolicy,
+  featureUsageService: FeatureUsageService,
+  endpointMetadataService: EndpointMetadataService
+) {
+  if (
+    !newPackagePolicy?.id ||
+    !newPackagePolicy?.inputs ||
+    !newPackagePolicy.inputs[0]?.config?.policy?.value
+  ) {
+    return;
+  }
+
+  const newPolicyConfig = newPackagePolicy.inputs[0].config?.policy?.value as PolicyConfig;
+
+  const currentPackagePolicy = await endpointMetadataService.getFleetEndpointPackagePolicy(
```

Comment:
can you explain what this function is doing by reaching back to Fleet and getting the endpoint package again?

I'm having a hard time understanding the code because the types for `newPackagePolicy` is set to `NewPackagePolicy`, so in my mind the policy is not created yet.

Reply:
This portion of the code is only triggers on policy update. So here I'm trying to compare the updated policy (`newPackagePolicy`) to the current one to determine if we're going from mode off to mode protect/detect. Since the current one isn't provided in the request, I had to fetch it.

Reply:
ahh. ok. got it. 

Could you add that info. with a command above this "for the future us" ? 😄